docs/3.0/develop/transactions.mdx

```diff
@@ -119,3 +125,61 @@ def confirmation(transaction):
     print("committing a record now using the task's cache key!")
 ```
 </Tip>
+
+### Access data within transactions
+
+Key-value pairs can be set within a transaction and accessed elsewhere within the transaction, including within the `on_rollback` hook.
+
+The code below shows how to set a key-value pair within a transaction and access it within the `on_rollback` hook and the `quality_test` task: 
+
+```python
+import os
+from time import sleep
+
+from prefect import task, flow
+from prefect.transactions import get_transaction, transaction
+
+
+@task
+def write_file(contents: str):
+    "Writes to a file."
+    with open("side-effect.txt", "w") as f:
+        f.write(contents)
+
+
+@write_file.on_rollback
+def del_file(transaction):
+    "Deletes file."
+    os.unlink("side-effect.txt")
+    txn1 = get_transaction()      # get the transaction object
+    print(txn1.get("contents"))   # prints "hello world"
+
+
+@task
+def quality_test():
+    "Checks contents of file."
+    with open("side-effect.txt", "r") as f:
+        data = f.readlines()
+
+    if len(data) < 2:
+        txn2 = get_transaction()      # get the transaction object
+        raise ValueError(f"Not enough data! {(txn2.get("contents"))}") 
```

Comment:
this isn't good code practice to display a different object than the one that got checked - I get why you showed it but I think if we want to show `get_transaction` within a task we should find a different example

Reply:
Yeah, I'd like to take out the `get_transaction` call. As soon as the issue with the[ order of setting transaction key-value pairs](https://github.com/PrefectHQ/prefect/issues/15309) is resolved, I'd get rid of that line.  I saw this current example as a temporary situation. I could just set the pair earlier, but didn't want someone to get surprised by the bug. 

Happy to change to whatever you think is preferable.

Reply:
Updated to just set it earlier in the latest pushed code.